### DISCUSSION TITLE
How to create a dataset from our own data

### DISCUSSION THREAD

**Discussion by zarkoivkovicc** [neuraloperator/neuraloperator#569] [Q&A]
I have a dataset as a numpy array with shape (b,t,g) with b being number of simulated trajectories, t number of time steps per trajectory and g number of spacial point on the grid (1D). How can I make the dataset and dataloaders that can be used with Trainer and FNO model?

**Comment by alundilong** [comment]
I think burger1D problem is very close to your problem, check the data format with following code could be helpful.
```
import torch
data=torch.load('burgers_train_16.pt')
data['x'].shape # output: torch.Size([800, 16]) where 16 is number of points
data['y'].shape # output: torch.Size([800, 17, 16]) where 17 should be number of timestep
```

**Comment by dhpitt** [comment]
Moving to a discussion
